### DISCUSSION TITLE
falcon_client.Submit(body=body) fails with invalid input

### DISCUSSION THREAD

**Discussion by bp4151** [CrowdStrike/falconpy#187] [Q&A] [answered]
Hey folks,

I am trying to submit a file after upload and am getting a 400 invalid input message. I have followed both the docs at 
[https://github.com/jshcodes/falconpy/wiki/FalconX-Sandbox#submit](https://github.com/jshcodes/falconpy/wiki/FalconX-Sandbox#submit)
and 
[https://assets.falcon.crowdstrike.com/support/api/swagger.html#/falconx-sandbox/Submit](https://assets.falcon.crowdstrike.com/support/api/swagger.html#/falconx-sandbox/Submit)

I am running the following
falconpy version 0.5.1
python version 3.9.1

My code is as follows

```
def submit_file(
        falcon_client: object,
        sha256: str,
        environment_id: int) -> dict:
    body = {
        "sandbox": [{
            "environment_id": environment_id,
            "sha256": sha256
        }],
        "user_tags": []
    }

    falcon_client.headers["Content-Type"] = "application/json"

    resp = falcon_client.Submit(body=body)
    print(resp)
    return resp
    
submit_result = submit_file(falcon_client=falcon_client, sha256=sha256, environment_id=args.EnvironmentId)
        submit_errors = submit_result.get('body').get('errors', [])
        if len(submit_errors) > 0:
            for error in submit_errors:
                print(error)
                
                ===> error = {"code": 400, "message": "Invalid user input"} <===
                
        else:
            print(submit_result)    

```

If someone could please help me out with this, I'd really appreciate it.

**Comment by bp4151** [comment, marked as answer]
Please ignore this. I figured it out. The API doesn't like the empty user_tags array and for some reason, my environment_id was coming in as a string instead of an int.

**Reply by mccbryan3** [reply]
Glad you got this figured out @bp4151 . I have been running the function and could not reproduce the error.
